DataDog/datadog-agent
# [pkg/collector] Reorder platform initialization on collector

### What does this PR do?

- Reorder `initializePlatform` call so that it happens as soon as possible during the rtloader's initialization.

### Motivation

- #5001 introduced the `windows_use_pythonpath` option to fix an issue where rtloader would crash while loading an incorrect Python version. However, in order to completely fix the issue that PR intended to address (which happens on Windows 2012 R2) we need to call the `initializePlatform` function  before the `make2`/`make3` functions.

### Describe your test plan

- Enable crash handling and check that the Agent runs correctly on Linux/macOS.
- On a Windows 2012 R2 machine install Python 2 and set the `PYTHONPATH` environment variable to point to this Python version. On the Agent configuration file set `python_version: 3`.  Manually run the Agent and check that it runs correctly.
- Repeat the previous steps but setting `windows_use_pythonpath` to true. Manually run the Agent and check that it throws an error similar to:

```
INFO | (pkg/collector/python/init.go:303 in Initialize) | Initializing rtloader with python3 C:\Program Files\Datadog\Datadog Agent\embedded3

Python path configuration:
  PYTHONHOME = 'C:\Program Files\Datadog\Datadog Agent\embedded3'
  PYTHONPATH = 'C:\Python27\Lib;C:\Python27\DLLs;'
  program name = 'python'
  isolated = 0
  environment = 1
  user site = 1
  import site = 1
  sys._base_executable = 'C:\\Program Files\\Datadog\\Datadog Agent\\bin\\agent.exe'
  sys.base_prefix = 'C:\\Program Files\\Datadog\\Datadog Agent\\embedded3'
  sys.base_exec_prefix = 'C:\\Program Files\\Datadog\\Datadog Agent\\embedded3'
  sys.executable = 'C:\\Program Files\\Datadog\\Datadog Agent\\bin\\agent.exe'
  sys.prefix = 'C:\\Program Files\\Datadog\\Datadog Agent\\embedded3'
  sys.exec_prefix = 'C:\\Program Files\\Datadog\\Datadog Agent\\embedded3'
  sys.path = [
    'C:\\Python27\\Lib',
    'C:\\Python27\\DLLs',
    'C:\\Python27\\Lib\\lib-tk',
    'C:\\Program Files\\Datadog\\Datadog Agent\\embedded3\\python38.zip',
    'C:\\Program Files\\Datadog\\Datadog Agent\\embedded3\\DLLs',
    'C:\\Program Files\\Datadog\\Datadog Agent\\embedded3\\lib',
    'C:\\Program Files\\Datadog\\Datadog Agent\\bin',
  ]
Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding
Python runtime state: core initialized
  File "C:\Python27\Lib\encodings\__init__.py", line 123
    raise CodecRegistryError,\
                            ^
SyntaxError: invalid syntax

Current thread 0x0000089c (most recent call first):
<no Python frame>
```
